### PR TITLE
Export Service

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Added Service to export data from a DAO into a given format.
2. Added ExportDriverRegister
3. Added ExportDriver interface
4. Added JSONDriver, XMLDriver, CSVDriver. The last two being unimplemented

### CODE REVIEW

File: src/foam/nanos/client/Client.js [hunk 1]
```diff
@@ -28,6 +29,7 @@ foam.CLASS({
   exports: [
     'countryDAO',
     'cronDAO',
+    'exportExportDriverRegistryDAO',
```
Comment (kgrgreer) [inline src/foam/nanos/client/Client.js]: Seems like one too many 'exports' in name.

File: src/foam/nanos/export/ExportDriverRegistry.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.export',
+  name: 'ExportDriverRegistry',
+  documentation: '',
```
Comment (kgrgreer) [inline src/foam/nanos/export/ExportDriverRegistry.js]: Better to fill this in.

File: src/foam/nanos/export/JSONDriver.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+ foam.CLASS({
+   package: 'foam.nanos.export',
+   name: 'JSONDriver',
+   implements: [ 'foam.nanos.export.ExportDriver' ],
+
+   properties: [
+     {
+       class: 'foam.json.Outputer',
```
Comment (kgrgreer) [inline src/foam/nanos/export/JSONDriver.js]: class: 'FObjectProperty',
of: 'foam.json.Outputer'

File: src/foam/nanos/export/JSONDriver.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+       name: 'outputer',
+       factory: function() { return foam.json.PrettyStrict; }
+     }
+   ],
+
+   methods: [
+     function exportDAO(X, dao) {
+       return dao.select().then(function (sink) {
+         return outputer.stringify(sink.a);
```
Comment (kgrgreer) [inline src/foam/nanos/export/JSONDriver.js]: sink.array

sink.a triggers a deprecation warning to the console

File: src/foam/nanos/client/Client.js [hunk 5]
```diff
@@ -82,6 +84,21 @@ foam.CLASS({
       }
     },
 
+    // TODO: change to client DAO
+    {
+      name: 'exportExportDriverRegistryDAO',
```
Comment (kgrgreer) [inline src/foam/nanos/client/Client.js]: One too many exports here also.

Comment (kirkeaton) [reply src/foam/nanos/client/Client.js]: Thanks, removed.

File: src/foam/nanos/export/JSONDriver.js (hunk 3 above)
Comment (kirkeaton) [reply src/foam/nanos/export/JSONDriver.js]: Done

File: src/foam/nanos/export/JSONDriver.js (hunk 4 above)
Comment (kirkeaton) [reply src/foam/nanos/export/JSONDriver.js]: Done

File: src/foam/nanos/export/ExportDriverRegistry.js (hunk 2 above)
Comment (kirkeaton) [reply src/foam/nanos/export/ExportDriverRegistry.js]: Done

File: src/foam/nanos/client/Client.js (hunk 1 above)
Comment (kirkeaton) [reply src/foam/nanos/client/Client.js]: Done

**Pull request merged**